scripts/automation/release/version_patcher.py

```diff
@@ -0,0 +1,77 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+#pylint: disable=line-too-long
+
+import os
+import sys
+import fileinput
+
+class VersionPatcher(object):
```

Comment:
I have not seen a way to use an environment variable with `setuptools` or `setup.py`.
Using an environment variable in `setup.py` could mean that if I user has this environment variable set, and then they install a component, the wrong version could be potentially shown.

Other approaches - https://packaging.python.org/single_source_version/
Also - http://stackoverflow.com/a/2073599